spec/models/cfp_spec.rb

```diff
@@ -1,8 +1,24 @@
 require 'spec_helper'
 
 describe Cfp do
+  subject { create(:cfp) }
   let!(:conference) { create(:conference, end_date: Date.today) }
-  let!(:cfp) { build(:cfp, start_date: Date.today - 2, end_date: Date.today - 1, program_id: conference.program.id) }
+  let!(:cfp) { create(:cfp, start_date: Date.today - 2, end_date: Date.today - 1, program_id: conference.program.id) }
+
+  describe 'scope' do
+    describe '#for_events' do
+      it 'returns the cfp for events' do
+        expect(conference.program.cfps.for_events).to be_a Cfp
+        expect(conference.program.cfps.for_events.cfp_type).to eq('events')
+      end
+    end
+  end
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:cfp_type) }
+    it { is_expected.to validate_inclusion_of(:cfp_type).in_array(Cfp::TYPES) }
+    it { is_expected.to validate_uniqueness_of(:cfp_type).scoped_to(:program_id) }
```

Comment:
Just after a brief look try using validate_uniqueness_of(:cfp_type).scoped_to(:program_id).case_insensitive
Will review the whole PR soon.